### ISSUE TITLE
Question: How to implement an enumeration type

### ISSUE THREAD

**Issue by carstenmichel** [shurcooL/graphql#67]
Hello,
the private API that I want to access needs an enumeration type in one of the parameters.
https://graphql.org/learn/schema/#enumeration-types
Is there a way to implement this with this library?
Many thanks

**Comment by dmitshur** [comment]
Yes, enumeration types are supported. You can either define types and consts that correspond to the private API schema, for example:

```Go
type Episode string

const (
	EpisodeNewHope Episode = "NEWHOPE"
	EpisodeEmpire  Episode = "EMPIRE"
	EpisodeJedi    Episode = "JEDI"
)
```

And then use the type and individual values. Or you can use the const values directly.

Package [`githubv4`](https://github.com/shurcooL/githubv4), which builds on top of this general-purpose `graphql` package, defines Go identifiers for the GitHub API schema via code generation, see its [enum.go](https://github.com/shurcooL/githubv4/blob/234843c633fadff9694210ce7ab97948c9148a14/enum.go) file.
